### PR TITLE
Add call to forget matrix API for past session deletion

### PR DESCRIPTION
**Pull request by FadhlanR**
Ticket: CS-6349

### CODE REVIEW

File: packages/matrix/tests/room-creation.spec.ts [hunk 1]
```diff
@@ -185,6 +186,8 @@ test.describe('Room creation', () => {
 
   test('it can delete a room', async ({ page }) => {
     await login(page, 'user1', 'pass');
+    let roomsBeforeDeletion = await getRoomsFromSync('user1', 'pass');
```
Comment (lukemelia) [inline packages/matrix/tests/room-creation.spec.ts]: Is there a reason to pass the username and password if it has defaults?

File: packages/matrix/tests/room-creation.spec.ts [hunk 2]
```diff
@@ -235,6 +224,14 @@ test.describe('Room creation', () => {
     expect(newRoom).not.toEqual(room2);
     expect(newRoom).not.toEqual(room3);
     await assertRooms(page, [newRoom]);
+
+    // For asserting the result of the forget matrix API
+    let roomsAfterDeletion = await getRoomsFromSync('user1', 'pass');
```
Comment (lukemelia) [inline packages/matrix/tests/room-creation.spec.ts]: Do we need to call with user and pass since this is the default credentials?

**Review by lukemelia — APPROVED** [review]

**Review by burieberry — COMMENTED** [review]
Can you check if you are also seeing the following `M_FORBIDDEN` issue? If you don't see it, try again, it does appear most of the time for me, though.

Replication steps: Open AI panel > Delete a room from Past Sessions > refresh the page > try to open AI panel again

forget-room

**Comment by burieberry** [comment]
Something I tried in the past is a call to kick the bot out of the room before leaving and forgetting the room, however, I still saw the same error on refresh.

**Comment by FadhlanR** [comment]
> Something I tried in the past is a call to kick the bot out of the room before leaving and forgetting the room, however, I still saw the same error on refresh.

It seems that we are still receiving events from the rooms that the user has left. To address this issue, I have added a validation to prevent this error. Please review this commit [db202dd](https://github.com/cardstack/boxel/pull/1090/commits/db202dd4f8632f651e9edbe1cfb0306fc8da997a).

**Review by burieberry — APPROVED** [review]
This fixes the error that I had seen.

**Pull request merged**